input/elasticapm/internal/modeldecoder/rumv3/decoder.go

```diff
@@ -146,12 +153,14 @@ func DecodeNestedTransaction(d decoder.Decoder, input *modeldecoder.Input, batch
 		event.Transaction = &model.Transaction{ID: transaction.Transaction.ID}
 		event.Parent.ID = transaction.Transaction.ID // may be overridden later
 		event.Trace = transaction.Trace
-		*batch = append(*batch, event)
+		var out modelpb.APMEvent
+		event.ToModelProtobuf(&out)
+		*batch = append(*batch, &out)
 	}
 	spans := (*batch)[offset:]
 	for i, s := range root.Transaction.Spans {
 		if s.ParentIndex.IsSet() && s.ParentIndex.Val >= 0 && s.ParentIndex.Val < len(spans) {
-			spans[i].Parent.ID = spans[s.ParentIndex.Val].Span.ID
+			spans[i].Parent.Id = spans[s.ParentIndex.Val].Span.Id
```

Comment:
Are `Parent` and `Span` guaranteed to be non-nil, or should we be checking them here?

Reply:
Updated!